ballerina/client.bal

```
@@ -357,760 +358,425 @@ public isolated client class Client {
-        if response is json {
-            return response;
+    // Added APIs
+    
+    # Retrieves the list of individual records that have been deleted within the given timespan.
+    #
+    # + sObjectName - SObject reference
+    # + startDate - Start date of the timespan.
+    # + endDate - End date of the timespan.
+    # + return - `DeletedRecordsResult` record if successful or else `error`
+    isolated remote function getDeleted(string sObjectName, time:Civil startDate, time:Civil endDate) 
+        returns DeletedRecordsResult|error {
+        string path = utils:prepareUrl([API_BASE_PATH, SOBJECTS, sObjectName, DELETED]);
+        string finalUrl = utils:addQueryParameters(path,{'start: check time:civilToString(removeDecimalPlaces(startDate)), 
+            end: check time:civilToString(removeDecimalPlaces(endDate))});
+        return check self.salesforceClient->get(finalUrl);
+    }
+
+    # Retrieves the list of individual records that have been deleted within the given timespan.
+    #
+    # + sObjectName - SObject reference
+    # + startDate - Start date of the timespan.
+    # + endDate - End date of the timespan.
+    # + return - `UpdatedRecordsResults` record if successful or else `error`
+    isolated remote function getUpdated(string sObjectName, time:Civil startDate, time:Civil endDate) 
+        returns UpdatedRecordsResults|error {
+        string path = utils:prepareUrl([API_BASE_PATH, SOBJECTS, sObjectName, UPDATED]);
+        string finalUrl = utils:addQueryParameters(path,{'start: check time:civilToString(removeDecimalPlaces(startDate)), 
+            end: check time:civilToString(removeDecimalPlaces(endDate))});
+        return check self.salesforceClient->get(finalUrl);
+    }
+
+    # Get the password information
+    #
+    # + userId - User ID
+    # + return - `boolean` if successful or else `error`
+    isolated remote function getPasswordInfo(string userId) returns boolean|error {
```

If this is to check only password expiry, shall we name as `isPasswordExpired`? 
Do we really need to return an error?

---

in cases of casting and network related errors, we need to return one right?